libkbfs/block_id.go

```diff
@@ -56,3 +60,21 @@ func (id BlockID) MarshalBinary() (data []byte, err error) {
 func (id *BlockID) UnmarshalBinary(data []byte) error {
 	return id.h.UnmarshalBinary(data)
 }
+
+// MarshalJSON implements the encoding.json.Marshaler interface for
+// BlockID.
+func (id BlockID) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("\"%s\"", id)), nil
+}
+
+// UnmarshalJSON implements the encoding.json.Unmarshaler interface
+// for BlockID.
+func (id BlockID) UnmarshalJSON(s []byte) error {
+	blockIDStr := strings.Trim(string(s), "\"")
```

Comment:
Here, too, rather than parsing the JSON yourself, maybe

```
func ... UnmarshalJSON(data []byte) ... {
  var s string
  err := json.Unmarshal(data, &s)
  ...
  newID, err := BlockIDFromString(s)
  ...
```